File: models/Post.js
const Post = (sequelize, DataTypes) => {
  const post = sequelize.define(
    'Post',
    {
      id: { type: DataTypes.INTEGER, primaryKey: true },
      title: DataTypes.STRING,
      content: DataTypes.STRING,
      userId: { type: DataTypes.INTEGER, foreignKey: true },
      published: { type: DataTypes.DATE, defaultValue: Date.now() },
      updated: { type: DataTypes.DATE, defaultValue: Date.now() },
    },

    {
      timestamps: false,
    },
  );

  post.associate = (models) => {
    post.belongsTo(models.User, {
      foreignKey: 'id', as: 'user_id',
    });
  };

  return post;
};

module.exports = Post;

File: blog_api/lib/blog_api/post.ex
defmodule BlogApi.Post do
  use Ecto.Schema
  import Ecto.Changeset

  alias BlogApi.User

  @primary_key {:id, Ecto.UUID, autogenerate: true}
  @foreign_key_type Ecto.UUID

  schema "posts" do
    field(:title, :string)
    field(:content, :string)
    belongs_to(:user, User)
    timestamps(inserted_at: :published, updated_at: :updated)
  end

  @required_params [:title, :content, :user_id]
  def changeset(params) do
    %__MODULE__{}
    |> cast(params, @required_params)
    |> validate_required(@required_params)
  end
end
